apps/molgenis-components/src/client/client.ts

```diff
@@ -451,3 +401,57 @@ const toFormData = (rowData: IRow) => {
 
   return formData;
 };
+
+async function convertRowToPrimaryKey(
+  myAxios: Axios,
+  row: IRow,
+  tableName: string,
+  schemaName?: string,
+  useCache: boolean = true
+): Promise<Record<string, any>> {
+  async function getKeyValue(
+    cellValue: any,
+    column: IColumn,
+    schemaName: string
+  ) {
+    if (typeof cellValue === "string") {
+      return cellValue;
+    } else {
+      if (column.refTable) {
+        return await convertRowToPrimaryKey(
+          myAxios,
+          cellValue,
+          column.refTable,
+          schemaName,
+          useCache
+        );
+      }
+    }
+  }
+
+  const schema = await fetchSchemaMetaData(myAxios, schemaName, useCache);
+  const tableMetadata = schema.tables.find(
```

Comment:
I feel there is a lot of logic below. Is this too much overhead or if this okay? Because in principle this will now be repeated for every row. What we could do instead is to apply this to a list of rows and then it is only one time. But maybe we should not care now and simply finish this major improvement.

Reply:
not use what part of de code you mean by 'logic below'?